core/src/main/java/com/datastax/oss/driver/internal/core/os/Native.java

```diff
@@ -15,67 +15,32 @@
  */
 package com.datastax.oss.driver.internal.core.os;
 
-import com.datastax.oss.driver.internal.core.util.Reflection;
-import java.lang.reflect.Method;
-import jnr.ffi.LibraryLoader;
 import jnr.ffi.Platform;
-import jnr.ffi.Pointer;
-import jnr.ffi.Runtime;
-import jnr.ffi.Struct;
-import jnr.ffi.annotations.Out;
-import jnr.ffi.annotations.Transient;
-import jnr.posix.POSIXFactory;
-import jnr.posix.util.DefaultPOSIXHandler;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /** A gateway to perform system calls. */
 public class Native {
 
-  private static final Logger LOG = LoggerFactory.getLogger(Native.class);
+  private static NativeImpl impl = new JnrNativeImpl();
```

Comment:
The key to the refactoring of this class: I'm laying the groundwork here for JAVA-2663 as well.  It's fairly easy to see how a GraalNativeImpl would fit in here.

Reply:
How do you plan to parameterize it and inject different implementations? We need to move it higher in the call chain (here probably constructor injection) and pass it from somewhere? Maybe parameterized `DriverContext`?

Reply:
I'm envisioning something like the following:

```java
  private static NativeImpl impl =
          /* Hope is that Graal build-time analysis will detect this case and follow the right path
          * (i.e. exclude JnrNativeImpl from consideration in the build).  Obviously this needs to be
          * tested.
          *
          * Constant values here come from org.graalvm.nativeimage.ImageInfo and are guaranteed not to change
          * in future releases.  Use the string literals here in order to avoid pulling in an import of GraalVM
          * code in this class. */
          System.getProperty("org.graalvm.nativeimage.imagecode").equals("buildtime") ?
            new GraalNativeImpl() :
              new JnrNativeImpl();
```

Obviously that'll need to be tested in a real case before we call it good.  That work will be handled in JAVA-2663.